src/amo/components/SearchResult/index.js

```diff
@@ -202,39 +230,44 @@ export class SearchResultBase extends React.Component<InternalProps> {
     );
   }
 
+  onClickResult = () => {
+    const { addon, addonInstallSource, clientApp, history, lang } = this.props;
+
+    if (addon) {
+      history.push(
+        `/${lang}/${clientApp}${this.getAddonLink(addon, addonInstallSource)}`,
+      );
+    }
+  };
+
   render() {
-    const { addon, addonInstallSource } = this.props;
+    const { addon } = this.props;
 
     const result = this.renderResult();
     const resultClassnames = makeClassName('SearchResult', {
       'SearchResult--theme': addon && isTheme(addon.type),
       'SearchResult--persona': addon && addon.type === ADDON_TYPE_THEME,
     });
 
-    let item = result;
-    if (addon) {
-      let linkTo = `/addon/${addon.slug}/`;
-      if (addonInstallSource) {
-        linkTo = addQueryParams(linkTo, { src: addonInstallSource });
-      }
-      item = (
-        <Link to={linkTo} className="SearchResult-link">
-          {result}
-        </Link>
-      );
-    }
-
-    return <li className={resultClassnames}>{item}</li>;
+    return (
+      // eslint-disable-next-line max-len
+      // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions, jsx-a11y/click-events-have-key-events
```

Comment:
These are helpful lint errors but I think it's fine to disable them. You could add a comment saying that the link is still keyboard accessible because an actual anchor tag was added to the `h2`. I don't think it's worth adding `onKeyUp` like one of the lint rules is asking you to.